File: android_plugins/notification/src/main/java/com/ironsrc/notify/Scheduler.java
package com.ironsrc.notify;

import android.app.Activity;
import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.Context;

class Scheduler {
    private Scheduler() {

    }

    static Scheduler getInstance() {
        return new Scheduler();
    }

    void schedule(Context context, PendingIntent intent, Scheduler.Options options) {
        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Activity.ALARM_SERVICE);
        alarmManager.cancel(intent);

        alarmManager.set(options.alarmType,
                System.currentTimeMillis() + options.triggerAtMillis,
                intent);
    }

    void cancel(Context context, PendingIntent intent) {
        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Activity.ALARM_SERVICE);
        alarmManager.cancel(intent);
    }

    static class Options {
        long triggerAtMillis;
        int alarmType;

        Options(long triggerAtMillis, int alarmType) {
            this.triggerAtMillis = triggerAtMillis;
            this.alarmType = alarmType;
        }
    }

}

File: android_plugins/notificationKt/src/main/java/com/ironsrc/notify/Scheduler.kt
package com.ironsrc.notify

import android.app.Activity
import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context


open class Scheduler private constructor(context: Context) {
    init {
        // Init using context argument
    }

    companion object : SingletonHolder<Scheduler, Context>(::Scheduler)
    data class Options(val triggerAtMillis: Long, val alarmType: Int = AlarmManager.RTC_WAKEUP)

    fun schedule(context: Context, intent: PendingIntent, options: Scheduler.Options) {
        val alarmManager = context.getSystemService(Activity.ALARM_SERVICE) as AlarmManager
        alarmManager.cancel(intent)

        alarmManager.set(options.alarmType,
                System.currentTimeMillis() + options.triggerAtMillis,
                intent)
    }

    fun cancel(context: Context, intent: PendingIntent) {
        val alarmManager = context.getSystemService(Activity.ALARM_SERVICE) as AlarmManager
        alarmManager.cancel(intent)
    }

}